src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -866,11 +866,6 @@ bool PSParallelCompact::initialize() {
   ParallelScavengeHeap* heap = ParallelScavengeHeap::heap();
   MemRegion mr = heap->reserved_region();
 
-  // Was the old gen get allocated successfully?
-  if (!heap->old_gen()->is_allocated()) {
-    return false;
-  }
-
```

Comment:
In principle this acts an assert, so maybe either adding one, or a comment like "// Memory reservation must have been successful if we reach here" would maybe be nice. Or renaming this method as `initialize_aux[iliary]_data` or something would already indicate that this code is not supposed to initialize everything.

Reply:
Renamed and added an assertion that heap should be reserved.